s3-backed-core/src/main/java/com/bakdata/kafka/AbstractS3BackedConfig.java

```diff
@@ -157,7 +172,23 @@ private boolean enablePathStyleAccess() {
         final String accessKey = this.getPassword(S3_ACCESS_KEY_CONFIG).value();
         final String secretKey = this.getPassword(S3_SECRET_KEY_CONFIG).value();
         if (isEmpty(accessKey) || isEmpty(secretKey)) {
-            return Optional.empty();
+            final String roleExternalId = this.getString(S3_ROLE_EXTERNAL_ID_CONFIG);
```

Comment:
In order to make this less nested, I think it would be better to have it like this:
```
if (!empty(accessKey) && !empty(secretKey)) {
  ...
  return ...
}
if (!empty(roleExternalId) && !empty(roleArn) && !empty(roleSessionName)) {
  ...
  return ...
}
return Optinal.empty()

Reply:
Sure thing!